snapshot-labs/sx-monorepo
# feat: support spaces follow and unfollow

### Summary



Closes partially #282

This PR add write support for the space's Follow/Unfollow feature.

For this PR scope, a follow and unfollow action will require a signing from your connected wallet.

### How to test

1. Connect with a wallet (Evm/Starknet)
2. Follow and unfollow some offchain and onchain spaces (you can only follow mainnet/testnet spaces depending on your env)
3. Followed spaces should appear in the sidebar
4. Confirm that the actions was successful by reloading the page: followed spaces should be persisted, and in the same order
5. When following more than 25 spaces, it should thrown an error

### Notes

- Can only follow EVM spaces for now, `sn` spaces still not supported, as sequencer is not using snapshotx api, but thegraph, due to https://github.com/checkpoint-labs/checkpoint/issues/296

### To-Do

- Coming in a later PR will be support for "session key", removing the need for message signing on each follow and unfollow request.
- [x] Waiting for https://github.com/snapshot-labs/snapshot.js/pull/995 to be released, and used by sequencer
- [x] Waiting for https://github.com/snapshot-labs/snapshot-sequencer/pull/351 so we can follow onchain spaces from sequencer
- [x] Change base to master once https://github.com/snapshot-labs/sx-monorepo/pull/288 is merged
